### PR TITLE
feat: pods table component

### LINKED ISSUE #4842: Move Pods page to table component
### Is your enhancement related to a problem? Please describe

We have a new table component, Pods page should switch.

### Describe the solution you'd like

The Pods page should be based on the new table component, and most columns should be sortable.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Switches the Pods page to the new table component. Columns and actions are extracted as expected. All the columns are sortable.

Table component allows update events to be passed through so that pod actions (or other future table actions) can trigger the UI to refresh.

### Screenshot / video of UI

Before:
Screenshot 2023-12-18 at 1 59 06 PM

After:
Screenshot 2023-12-18 at 1 57 21 PM

### What issues does this PR fix or reference?

Fixes #4842.

### How to test this PR?

Use pods list: start/stop/create pods, try container links and actions.

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
PodColumnActions.svelte looks untested (missing file ?)

File: packages/renderer/src/lib/table/Table.svelte [hunk 1]
```diff
@@ -189,7 +189,8 @@ function setGridColumns() {
             {#if column.info.renderer}
               <svelte:component
                 this="{column.info.renderer}"
-                object="{column.info.renderMapping?.(object) ?? object}" />
+                object="{column.info.renderMapping?.(object) ?? object}"
+                on:update />
```
Comment (benoitf) [inline packages/renderer/src/lib/table/Table.svelte]: I think this change should not be part of this PR that is focusing on pods 

(also it is missing a test case)

Comment (deboer-tim) [reply packages/renderer/src/lib/table/Table.svelte]: Separated out into #5345 (with test).

**Comment by deboer-tim** [comment]
Rebased to pick up table on:update change (PR #5345) and added action column test.

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
I think it can be merged, was enough time for review

**Comment by deboer-tim** [comment]
Merged, thanks.
